error C2338 when building with double precision

#### System Information
- Ogre Version: 1.12.5
- Operating System / Platform: Windows / x64
- Compiler: MSVC 2015

#### Detailed description
Build fails when compiling OgreInstanceBatch.cpp with double precision.

#### Build log
```
E:\Devel\Ogre\v1.12.5-master\ogre-master\OgreMain\src\OgreInstanceBatch.cpp(534): warning C4244: 'initializing': conversion from 'const Ogre::Real' to 'float', possible loss of data
e:\devel\ogre\v1.12.5-master\ogre-master\ogremain\include\OgreVector.h(61): error C2338: must have at least 4 dimensions
```

--- a/OgreMain/include/OgreVector.h
+++ b/OgreMain/include/OgreVector.h
@@ -301,7 +301,7 @@ namespace Ogre
         you interpret the values.
     */
     template<int dims, typename T>
-    class _OgreExport Vector : public VectorBase<dims, T>
+    class Vector : public VectorBase<dims, T>
     {
     public:
         using VectorBase<dims, T>::ptr;

